### DISCUSSION TITLE
Add built-in keyboard

### DISCUSSION THREAD

**Discussion by litmich** [Kunzisoft/KeePassDX#1830] [Ideas]
Magi keyboard really lacks some basic keyboard functionality, for example 3 use cases:
1. Filling a form with few fields stored with magi keyboard, then if some fields are not stored in the vault and need to be filled manually - need to switch to another keyboard and back to magi keyboard.
2. Using vault locked by pin without biometrics. Invoke magi keyboard, switch to another keyboard for pin entry, reopen magi keyboard.
3. Search entry.
Open magi keyboard, to search an entry manually switch to another keyboard, reopen magikeyboard.

All this results in too much manual jungling between the keyboards, especially if all 3 use cases encountered in sequence. Adding some basic alphanumeric built in keyboard would greatly smooth the process.
Thanks.

**Comment by J-Jamet** [comment]
1. https://github.com/Kunzisoft/KeePassDX/issues/311
2. https://github.com/Kunzisoft/KeePassDX/issues/311
3. https://github.com/Kunzisoft/KeePassDX/issues/311
:)
